attest/verifier/types/src/verification.rs

```diff
@@ -349,4 +379,72 @@ mod tests {
             "VerificationReport { sig: deadbeefcafe, chain: [abcd, cdef, 1234], http_body: \"some_body\" }"
         );
     }
+
+    #[test]
+    fn enclave_report_data_contents_digest() {
+        let nonce: QuoteNonce = [0x1u8; 16].into();
+        let key_bytes = [0x22u8; 32];
+        let key: X25519Public = key_bytes.as_slice().try_into().expect("bad key");
+        let custom_identity = [0x33u8; 32];
+        let report_data_1 =
+            EnclaveReportDataContents::new(nonce.clone(), key.clone(), custom_identity);
+
+        let report_data_2 =
+            EnclaveReportDataContents::new(nonce.clone(), key.clone(), custom_identity);
+
+        let digest_1 = report_data_1.digest32::<MerlinTranscript>(b"");
+        let digest_2 = report_data_2.digest32::<MerlinTranscript>(b"");
+        assert_eq!(digest_1, digest_2);
+
+        let mut modified_nonce = nonce.clone();
+        let nonce_bytes: &mut [u8] = modified_nonce.as_mut();
+        nonce_bytes[0] += 1;
+        let modified_nonce_report_data =
+            EnclaveReportDataContents::new(modified_nonce, key.clone(), custom_identity);
+
+        let modified_nonce_digest = modified_nonce_report_data.digest32::<MerlinTranscript>(b"");
+        assert_ne!(digest_1, modified_nonce_digest);
```

Comment:
I'm not a fan of workflow or recipe tests, where one sets up the base case, then toggles a lever and re-asserts something. I used a workflow test here as it felt a bit heavy to repeat the 10 line setup in mutliple test cases and I'm not sure functionalizing it would be much better.

That said if one can think of a clearer or more concise way to test, I'm open for suggestions.

Reply:
I think it's fine in this case